tracer/test/Datadog.Trace.Tests/Tagging/TagsListTests.cs

```diff
@@ -116,35 +113,31 @@ public void Serialization(bool topLevelSpan)
             span.SetTag(Tags.Env, "Overridden Environment");
             span.SetMetric(Metrics.SamplingLimitDecision, 0.75);
 
-            for (int i = 0; i < 15; i++)
+            const int customTagCount = 15;
+
+            for (int i = 0; i < customTagCount; i++)
             {
                 span.SetTag(i.ToString(), i.ToString());
             }
 
-            for (int i = 0; i < 15; i++)
+            for (int i = 0; i < customTagCount; i++)
             {
                 span.SetMetric(i.ToString(), i);
             }
 
-            var buffer = new byte[0];
-
-            // use vendored MessagePack to serialize
-            var resolver = SpanFormatterResolver.Instance;
-            Vendors.MessagePack.MessagePackSerializer.Serialize(ref buffer, 0, span, resolver);
-
-            // use nuget MessagePack to deserialize
-            var deserializedSpan = global::MessagePack.MessagePackSerializer.Deserialize<MockSpan>(buffer);
+            var deserializedSpan = SerializeSpan(span);
 
-            // For top-level spans, there is one tag added during serialization
-            Assert.Equal(topLevelSpan ? 17 : 16, deserializedSpan.Tags.Count);
+            // For top-level spans, there are 3 tags added during serialization: "env", "language", and "runtime-id".
+            // Otherwise, there are 2 tags added during serialization: "env" and "language".
+            Assert.Equal(topLevelSpan ? customTagCount + 3 : customTagCount + 2, deserializedSpan.Tags.Count);
```

Comment:
_sigh_ what does top-level mean here? "local root"? Or "service-entry"?

Reply:
Service-entry span :) I rewrote the comments in `TagsList` (37678b51e781891c896c6783cff125eba8128085) and refactored the tests (e50042d330319870b9797147ed5b26f0b44b3f9f).